### ISSUE TITLE
upgrade firebase related packages / full null safety support

### ISSUE THREAD

**Issue by ghost** [stelynx/collectio#193]
Hi @campovski ,

Hope you are all well !

I played with tour project and it would be nice to upgrade all packages and have a full support of null-safety ^^

Cheers,
Luc

**Comment by campovski** [comment]
Yes, I agree. This project has been neglected for too long, not even released to App Store and Google Play, however it is more than ready to be (except for null safety).

I remember I played around a bit with adding null safety to this project some time ago, however a lot of logic depends on types being nullable it is a complete mess to upgrade it. If you would like, I can take some time off from my current project and focus on this 😊
